### PR TITLE
Disallow `unknown` and bump version to 3.2.2

### PR DESCRIPTION
**Pull request by aaronjeline**
## Description of changes
Backport of #1101  and bumps version to 3.2.2
## Issue #, if available
#1096 
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [X] A bug fix or other functionality change requiring a patch to `cedar-policy`.

I confirm that this PR (choose one, and delete the other options):

- [X] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [X] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

### CODE REVIEW

**Review by khieta — COMMENTED** [review]
As discussed elsewhere, the integration test failure in `cedar-drt` is expected because the CI is borked 😞 I'll file an issue shortly

File: cedar-policy/src/integration_testing.rs [hunk 1]
```diff
@@ -52,6 +52,7 @@ pub struct JsonTest {
     pub schema: String,
     /// Whether the given policies are expected to pass the validator with this
     /// schema, or not
+    #[serde(rename = "ShouldValidate")]
```
[the remark this answers is not in the document]
Comment (khieta) [reply cedar-policy/src/integration_testing.rs:55]: Nvm, I know why this didn't break things: this part of the code is deprecated starting in 3.2.0 so it's not actually used for testing.

[the remark this answers is not in the document]
Comment (aaronjeline) [reply cedar-policy/src/integration_testing.rs:55]: Not having it _did_ break CI?

[the remark this answers is not in the document]
Comment (khieta) [reply cedar-policy/src/integration_testing.rs:55]: Hm, it shouldn't have?

File: cedar-policy/CHANGELOG.md [hunk 2]
```diff
@@ -3,3 +3,568 @@
 The changelog for all the release branches of `cedar-policy` is maintained on
 the `main` branch. You can view the most up-to-date changelog
 [here](https://github.com/cedar-policy/cedar/blob/main/cedar-policy/CHANGELOG.md).
+
+## [3.2.2] - TBA
+
+### Changed
+
+- `unknown()` is no longer a valid extension function if `partial-eval`
+  is not enabled as a feature.
```
Comment (khieta) [inline cedar-policy/CHANGELOG.md:12]: I also meant you should delete the rest of the added CHANGELOG text 🙂 It's annoying to keep it in sync with `main`.

**Review by khieta — APPROVED** [review]

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**